packages/backend/src/ee/services/ai/agents/tests/agent.integration.test.ts

```diff
@@ -256,4 +247,74 @@ describeOrSkip('agent integration tests', () => {
 
         expect(jsonDiffEvaluation.score).toBeGreaterThanOrEqual(0.5);
     });
+
+    it(
+        'should retrieve relevant context for a time-based query',
+        async () => {
+            if (!createdAgent) {
+                throw new Error('Agent not created');
+            }
+
+            const promptQueryText =
+                'Show me revenue by month from the orders data';
+
+            const { response, prompt } = await promptAgent(promptQueryText);
+
+            const toolCalls = await context
+                .db<DbAiAgentToolCall>('ai_agent_tool_call')
+                .leftJoin(
+                    'ai_agent_tool_result',
+                    'ai_agent_tool_call.tool_call_id',
+                    'ai_agent_tool_result.tool_call_id',
+                )
+                .where('ai_agent_tool_call.ai_prompt_uuid', prompt!.promptUuid)
+                .select('*');
```

Comment:
there is a    ` async getToolCallsAndResultsForPrompt(promptUuid: string) {` method that returns both tool calls and tool results with inner join. might be helpful to use here

Reply:
I want to use that method when we get to remove the feature flag for the _experimental_ find fields tool you've built 🙌 so will skip this suggestion for now 🙏